coredocument/validator.go

```diff
@@ -201,7 +202,7 @@ func documentRootValidator() documents.Validator {
 // re-calculates the signature and compares with existing one
 // assumes signing_root is already generated and verified
 // Note: this needs to used only before document is sent for signatures from the collaborators
-func readyForSignaturesValidator() documents.Validator {
+func readyForSignaturesValidator(ctxHeader *header.ContextHeader) documents.Validator {
```

Comment:
ctxHeader is a good thing as long as it doesn't replace everything we want to pass to a function with a single parameter. It can be a dramatic anti pattern in code since it wouldn't be obvious what you need to pass to a function by looking at its signature. Let's have a discussion on this change.